app/routes.py

```diff
@@ -1,2 +1,218 @@
-from flask import Blueprint
+from datetime import datetime
+from flask import Blueprint, json, make_response, request, jsonify
+from app import db
+from app.models.customer import Customer
+from app.models.video import Video
+from app.models.rental import Rental
+import os
+import requests
 
+customers_bp = Blueprint("customers", __name__, url_prefix="/customers")
+videos_bp = Blueprint("videos", __name__, url_prefix="/videos")
+rentals_bp = Blueprint("rentals", __name__, url_prefix="/rentals")
+
+                                       ### Crud Customers ###
+#decided to combine the post and get.
+
+@customers_bp.route("", methods=["GET", "POST"], strict_slashes = False)
+def customer_main_func():
+
+    if request.method == "POST":
+        customer_info = request.get_json()
+        
+        if "name" not in customer_info or "postal_code" not in customer_info or "phone" not in customer_info:
+            return make_response({"details":"invalid data"}, 400)
+        
+        new_customer = Customer(
+        name = customer_info["name"],
+        postal_code = customer_info["postal_code"],
+        phone = customer_info["phone"]
+        )
+        db.session.add(new_customer)
+        db.session.commit() 
+        
+        response = {
+            "id": new_customer.customer_id
+        }
+        return jsonify(response), 201
+
+    elif request.method == "GET":
+        customers = Customer.query.all()
+        response = []
+        for customer in customers:
+            response.append(customer.resp_json())
+        return jsonify(response), 200
+
+#change code to combine get, put, delete methods, easier to follow in my brain. Def went back to the style I had on task_list_api. 
+
+
+@customers_bp.route("/<customer_id>", methods=["GET", "PUT", "DELETE"], strict_slashes = False)
+def customer_id_main_func(customer_id):
+    customer = Customer.query.get_or_404(customer_id)
+
+
+    if not customer:
+        return jsonify({
+            "Customer info invalid. Try again!"
+        }, 404)
```

Comment:
you can add a description argument to `get_or_404()` that would look like this:
```suggestion
    customer = Customer.query.get_or_404(customer_id, description="Customer info invalid. Try again!")

    if not customer:
        return jsonify({
            "Customer info invalid. Try again!"
        }, 404)
```